addons/binding/org.openhab.binding.lgwebos/src/main/java/org/openhab/action/lgwebos/internal/LGWebOSAction.java

```diff
@@ -0,0 +1,235 @@
+/**
+ * Copyright (c) 2010-2017 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.action.lgwebos.internal;
+
+import java.awt.image.BufferedImage;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.net.URL;
+import java.util.Base64;
+import java.util.Collections;
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.concurrent.ArrayBlockingQueue;
+import java.util.concurrent.BlockingQueue;
+import java.util.stream.Collectors;
+
+import javax.imageio.ImageIO;
+
+import org.eclipse.smarthome.model.script.engine.action.ActionDoc;
+import org.eclipse.smarthome.model.script.engine.action.ActionService;
+import org.eclipse.smarthome.model.script.engine.action.ParamDoc;
+import org.openhab.binding.lgwebos.internal.discovery.LGWebOSDiscovery;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.connectsdk.core.AppInfo;
+import com.connectsdk.device.ConnectableDevice;
+import com.connectsdk.service.capability.CapabilityMethods;
+import com.connectsdk.service.capability.Launcher;
+import com.connectsdk.service.capability.TextInputControl;
+import com.connectsdk.service.capability.ToastControl;
+import com.connectsdk.service.capability.listeners.ResponseListener;
+import com.connectsdk.service.command.ServiceCommandError;
+import com.connectsdk.service.sessions.LaunchSession;
+import com.google.common.collect.Iterables;
+
+/**
+ * This class provides a rules action to start applications and send toast messages.
+ *
+ * @author Sebastian Prehn
+ * @since 2.1.0
+ */
+@Component(service = ActionService.class, immediate = true, name = "action.lgwebos")
+public class LGWebOSAction implements ActionService {
+    private static final Logger LOGGER = LoggerFactory.getLogger(LGWebOSAction.class);
```

Comment:
I thought about this for a while and it also says 'wherever possible'. Also following the link from that page it says:
`We used to recommend that loggers members be declared as instance variables instead of static. After further analysis, we no longer recommend one approach over the other.`

Maybe as long as methods are static, it makes sense for the logger to be static. There obviously are lots of static loggers all over the current code. What do you think?

Reply:
I know. found this also during the static code analysis. But to log form static action methods what else can I do?